### PR TITLE
PADV-1395 feat: bulk students enroll

### PR DESCRIPTION
**Pull request by AuraAlba**
### Ticket
https://agile-jira.pearson.com/browse/PADV-1395

### Description
Bulk students enrollment

### Changes made
Change input to textarea
Use backend response to get the emails validation
Update test

### Screenshots
bulkenrollment

### How to test
Clone the Institution Portal MFE
Run npm install.
Run npm run start
Go to http://localhost:1980/institution-portal/classes -> class detail page -> in menu actions select invite student to enroll

### CODE REVIEW

File: src/features/Classes/EnrollStudent/index.jsx [hunk 1]
```diff
@@ -47,20 +47,47 @@ const EnrollStudent = ({ isOpen, onClose, queryClassId }) => {
...
+        }
+        validEmails = [
+          ...validEmails,
+          email.identifier,
+        ];
+        return validEmails;
+      });
+
+      if (invalidEmails.length > 0) {
+        textToast = `The following email adresses are invalid: ${invalidEmails.join(' ')}\n`;
```
Comment (01001110J) [inline src/features/Classes/EnrollStudent/index.jsx:84]: What if only fails one email? I think the message for that scenario is not the best.

File: src/features/Classes/EnrollStudent/index.jsx [hunk 2]
```diff
@@ -47,20 +47,47 @@ const EnrollStudent = ({ isOpen, onClose, queryClassId }) => {
...
-        setToastMessage(decodeURIComponent(response?.data?.results[0]?.message));
+      if (messages?.data?.results[0]?.tags === 'error') {
+        setToastMessage(decodeURIComponent(messages?.data?.results[0]?.message));
         setShowToast(true);
 
         return onClose();
       }
 
-      setToastMessage('Email invite has been sent successfully');
+      validationEmailList.map(email => {
```
Comment (01001110J) [inline src/features/Classes/EnrollStudent/index.jsx:68]: Is necessary `.map` here?

I think this expression could be more simple, but you decide if you see it necessary.
```js
      validationEmailList.forEach(({ invalidIdentifier, identifier }) => {
        (invalidIdentifier ? invalidEmails : validEmails).push(identifier);
      });
```

File: src/features/Classes/EnrollStudent/index.jsx [hunk 3]
```diff
@@ -47,20 +47,47 @@ const EnrollStudent = ({ isOpen, onClose, queryClassId }) => {
...
+          email.identifier,
+        ];
+        return validEmails;
+      });
+
+      if (invalidEmails.length > 0) {
+        textToast = `The following email adresses are invalid: ${invalidEmails.join(' ')}\n`;
+      }
+      if (validEmails.length > 0) {
+        textToast += `Successfully enrolled and sent email to the following users: ${validEmails.join(' ')}`;
```
Comment (01001110J) [inline src/features/Classes/EnrollStudent/index.jsx:87]: What if I only enroll one single user? I think this message could be improved.

File: src/features/Classes/EnrollStudent/index.jsx [hunk 4]
```diff
@@ -113,12 +140,13 @@ const EnrollStudent = ({ isOpen, onClose, queryClassId }) => {
             <Form onSubmit={handleEnrollStudent}>
               <FormGroup controlId="studentInfo">
                 <Form.Control
-                  type="email"
+                  as="textarea"
```
Comment (01001110J) [inline src/features/Classes/EnrollStudent/index.jsx:143]: This new change make the distances between the text and its borders see weir, maybe a `min-heigth: 44px` could make see this more better.

image

File: src/features/Classes/EnrollStudent/index.jsx [hunk 5]
```diff
@@ -47,20 +47,47 @@ const EnrollStudent = ({ isOpen, onClose, queryClassId }) => {
...
+      });
+
+      if (invalidEmails.length > 0) {
+        textToast = `The following email adresses are invalid: ${invalidEmails.join(' ')}\n`;
+      }
+      if (validEmails.length > 0) {
+        textToast += `Successfully enrolled and sent email to the following users: ${validEmails.join(' ')}`;
+      }
+
+      setToastMessage(textToast);
```
Comment (01001110J) [inline src/features/Classes/EnrollStudent/index.jsx:90]: Something that does not convince to much to me is the error and success message are displayed in the same toast I would recommend to add more vertical space between each other at least.

**Review by 01001110J — APPROVED** [review]

**Review by anfbermudezme — APPROVED** [review]

**Pull request merged**